anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/AnomalyDetectorManager.java

```diff
@@ -67,6 +67,10 @@ public AnomalyDetector detectorFor(MappedMpoint mappedMpoint) {
             detector = factory.create(detectorUuid);
             detectors.put(detectorUuid, detector);
         }
+        PerformanceMonitor perfMonitor = new PerformanceMonitor();
+        MonitorDetector monitorDetector = new MonitorDetector(detector, perfMonitor);
+        // FIXME Should this be part of below classify method? [KS]
+        monitorDetector.classify(mappedMpoint);
```

Comment:
In fact, please change this detectorFor() method to accept a Metric instead of an Mpoint just to make this clearer. You'll have to pass in the mpoint.getMetric().